src/streamdiffusion/wrapper.py

```
@@ -446,18 +451,28 @@ def update_stream_params(
         seed_interpolation_method : Literal["linear", "slerp"]
             Method for interpolating between seed noise tensors, by default "linear".
         """
-        self.stream.update_stream_params(
+        new_stream = self.stream._param_updater.update_stream_params(
             num_inference_steps=num_inference_steps,
             guidance_scale=guidance_scale,
             delta=delta,
             t_index_list=t_index_list,
             seed=seed,
+            width=width,
+            height=height,
             prompt_list=prompt_list,
             negative_prompt=negative_prompt,
             interpolation_method=interpolation_method,
             seed_list=seed_list,
             seed_interpolation_method=seed_interpolation_method,
         )
+        # If a new stream was created (pipeline restart), update self.stream
+        if new_stream is not None:
+            self.stream = new_stream
```

Curious, when might this happen? I was assuming that `update_stream_params` would always be a quick in-place operation, maybe it'd be better to raise an exception and let the caller recreate the pipeline? Or would there be an advantage of recreating it internally here?

---

In `update_stream_params`, a `new_stream` is only created if a width or height is passed to update, shown [here](https://github.com/livepeer/StreamDiffusion/blob/52117d582f923208449da6940bfdb774464a6e5e/src/streamdiffusion/stream_parameter_updater.py#L135).

I'm not 100% sure if this is the best approach - open to suggestions. The old stream is properly cleared from memory if this happens.

---

But is it recreated more efficiently than recreating the whole wrapper for example? Does it re-use the engines already loaded in memory or something?